kubeflow-pipelines/common/components.py

```diff
@@ -241,3 +241,306 @@ def download_docling_models(
+    base_image=DOCLING_BASE_IMAGE,
+)
+def docling_chunk(
+    input_path: dsl.Input[dsl.Artifact],
+    output_path: dsl.Output[dsl.Artifact],
+    max_tokens: int = 1024,
+    merge_peers: bool = True,
+):
+    """
+    Chunk Docling documents using HybridChunker. Takes converted docling JSON files as input
+    and produces chunked JSON files with semantic chunks suitable for RAG.
+
+    Args:
+        input_path: Path to the input directory containing Docling JSON files
+        output_path: Path to the output directory for the chunked JSON files
+        max_tokens: Maximum number of tokens per chunk
+        merge_peers: Whether to merge smaller chunks at the same level
+    """
+    import json  # pylint: disable=import-outside-toplevel
+    from pathlib import Path  # pylint: disable=import-outside-toplevel
+
+    # HybridChunker = Docling's smart chunking class that combines:
+    # 1. Document structure awareness
+    # 2. Token-based splitting
+    from docling.chunking import HybridChunker  # pylint: disable=import-outside-toplevel
+    from docling_core.transforms.chunker.tokenizer.huggingface import (
+        HuggingFaceTokenizer,
+    )  # pylint: disable=import-outside-toplevel
+    from docling_core.types import DoclingDocument  # pylint: disable=import-outside-toplevel
+    from transformers import AutoTokenizer  # pylint: disable=import-outside-toplevel
+
+    # Convert KFP artifact paths to Path objects
+    input_path_p = Path(input_path.path)
+    output_path_p = Path(output_path.path)
+    output_path_p.mkdir(parents=True, exist_ok=True)
+
+    # Initialize tokenizer for HybridChunker (new API)
+    # Using a lightweight sentence-transformer model for tokenization
+    EMBED_MODEL_ID = "sentence-transformers/all-MiniLM-L6-v2"
+    tokenizer = HuggingFaceTokenizer(
+        tokenizer=AutoTokenizer.from_pretrained(EMBED_MODEL_ID),
+        max_tokens=max_tokens,
+    )
+
+    # Initialize Hybrid chunker with user-specified parameters
+    # tokenizer: The tokenizer wrapper to use for counting tokens (includes max_tokens)
+    # merge_peers: if true, smaller adjacent chunks will be merged together
+    chunker = HybridChunker(
+        tokenizer=tokenizer,
+        merge_peers=merge_peers,
+    )
+
+    # Find all JSON files in the input directory
+    json_files = list(input_path_p.glob("*.json"))
+    if not json_files:
+        print(f"docling-chunk: No JSON files found in {input_path_p}", flush=True)
+        return
+
+    print(
+        f"docling-chunk: processing {len(json_files)} files with max_tokens={max_tokens} and merge_peers={merge_peers}",
+        flush=True,
+    )
+
+    # Process each file
+    for json_file in json_files:
+        print(f"docling-chunk: processing {json_file}", flush=True)
+        with open(json_file, "r", encoding="utf-8") as f:
+            doc_data = json.load(f)
+
+        # parse the JSON data into a DoclingDocument object
+        doc = DoclingDocument.model_validate(doc_data)
```

Comment:
What if the JSON document is not a DoclingDocument here? What kind of exception would you expect to be thrown? It's good that you're running model_validate here but we need to use it handle cases of bad input.